site/src/components/VSCodeDesktopButton/VSCodeDesktopButton.tsx

```diff
@@ -0,0 +1,52 @@
+import Button from "@material-ui/core/Button"
+import { getApiKey } from "api/api"
+import { VSCodeIcon } from "components/Icons/VSCodeIcon"
+import React from "react"
+
+export interface VSCodeDesktopButtonProps {
+  userName: string
+  workspaceName: string
+  agentName?: string
+}
+
+export const VSCodeDesktopButton: React.FC<
+  React.PropsWithChildren<VSCodeDesktopButtonProps>
+> = ({ userName, workspaceName, agentName }) => {
+  const [loading, setLoading] = React.useState(false)
```

Comment:
Just a NIT: sometimes it's nice to `import {FC, PropsWithChildren, useState} from "react"` up top instead of importing all of React; this saves on compile time.

Reply:
Fixed!